ReportCardParser/src/reportcardparser/ReportCardStats.java

```diff
@@ -42,7 +66,33 @@ public static String GetBestSubject(ReportCard reportCard) {
      * @return The average mark of the passed-in subject
      */
     public static double SubjectAverage(ReportCard[] reportCards, String subject) {
-        // REPLACE ME WITH A REAL IMPLEMENTATION!
-        return 0.0;
+        
+        //sum of all the marks of the subject
+        double marksSum = 0;
+        
+        //ittorate through all of the reportCards
+        for (ReportCard card : reportCards) {
```

Comment:
What if reportCards is null (or the array entries are null)?  What if the subject is not found for a particular student?